api/src/blocks.rs

```diff
@@ -0,0 +1,167 @@
+    version::Version,
+};
+use std::cmp::min;
+
+use aptos_api_types::{
+    AccountData, Address, AsConverter, Error, HashValue, LedgerInfo, MoveModuleBytecode, Response,
+    TransactionId, TransactionOnChainData,
+};
+use aptos_types::{
+    account_config::AccountResource,
+    account_state::AccountState,
+    event::{EventHandle, EventKey},
+};
+
+use crate::param::Param;
+use anyhow::Result;
+use aptos_types::{
+    access_path::AccessPath, state_store::state_key::StateKey, transaction::Transaction,
+};
+use move_deps::move_core_types::{
+    identifier::Identifier,
+    language_storage::{ResourceKey, StructTag},
+    move_resource::MoveStructType,
+    value::MoveValue,
+};
+use std::convert::TryInto;
+use warp::{filters::BoxedFilter, Filter, Rejection, Reply};
+
+pub type BlockParam = Param<u64>;
+
+// GET /blocks/<address>
+pub fn get_block(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("blocks" / BlockParam)
+        .and(warp::get())
+        .and(context.filter())
+        .and_then(handle_get_block())
+        .with(metrics("get_block"))
+        .boxed()
+}
+
+// GET /genesis
+pub fn get_genesis_ledger_info(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("genesis")
+        .and(warp::get())
+        .and(context.filter())
+        .and_then(handle_get_genesis_ledger_info())
+        .with(metrics("get_genesis_ledger_info"))
+        .boxed()
+}
+
+async fn handle_get_block(
+    ledger_version: Option<LedgerVersionParam>,
+    context: Context,
+) -> Result<impl Reply, Rejection> {
+    fail_point("endpoint_get_block")?;
+    Ok(Block::new(ledger_version, context)?.account()?)
+}
+
+async fn handle_get_genesis_ledger_info(context: Context) -> Result<impl Reply, Rejection> {
+    fail_point("endpoint_get_genesis_ledger_info")?;
+    Ok(context.get_genesis_ledger_info()?)
+}
+
+pub struct BlockDescription {
+    starting_version: u64,
+    ending_version: u64,
+    num_transactions: u64,
+    hash: HashValue,
+}
+
+pub(crate) struct Block {
+    ledger_version: u64,
+    latest_ledger_info: LedgerInfo,
+    context: Context,
+}
+
+impl Block {
+    pub fn new(
+        ledger_version: Option<LedgerVersionParam>,
+        context: Context,
+    ) -> Result<Self, Error> {
+        let latest_ledger_info = context.get_latest_ledger_info()?;
+        let ledger_version = ledger_version
+            .map(|v| v.parse("ledger version"))
+            .unwrap_or_else(|| Ok(latest_ledger_info.version()))?;
+
+        if ledger_version > latest_ledger_info.version() {
+            return Err(Error::not_found(
+                "ledger",
+                TransactionId::Version(ledger_version),
+                latest_ledger_info.version(),
+            ));
+        }
+
+        Ok(Self {
+            ledger_version,
+            latest_ledger_info,
+            context,
+        })
+    }
+
+    /// Scans the DB
+    pub fn find_block(self) -> Result<BlockDescription, Error> {
```

Comment:
Make a API on DbReader similar to this, you will be able to get the block metadata txn version corresponding to an input version. (see how it calls the underlying `get_block_metadta()` which returns the version and the metadata

https://github.com/aptos-labs/aptos-core/blob/a87e1e0b448c6d884be7308f32fdc64a5a5ec360/storage/aptosdb/src/lib.rs#L1176